### DISCUSSION TITLE
Python3 removed - still having stuff in /usr/local/bin point to it

### DISCUSSION THREAD

**Discussion by duemme** [Homebrew/discussions#1331] [Getting started]
Hi,
I removed python3.9
In /usr/local/bin/ I still have: iptest, iptest3, ipython, ipython3, jupyter, jupiter-kernel, jupiter-kernelspec, jupiter-migrate, jupiter-run, jupyter-troubleshoot.
they look for python3.9 that has been removed.
Do they come with the brew installation?
Do I need to manually remove them?
thanks

**Comment by SMillerDev** [comment]
>  Do I need to manually remove them?

Yes, brew won't automatically remove all dependent formulae.

**Reply by duemme** [reply]
Thanks.
So all those binaries comes with the python installation? Where can I get a list of all the stuff installed by Brew while installing Python 3.9 and not removed by the "remove" argument?

**Reply by carlocab** [reply]
None of those were installed by `brew` when installing Python. You likely installed them using `pip3` if you were able to remove Python without `brew` complaining.

**Reply by duemme** [reply]
Ah ok. Maybe when I installed sypder-kernels via pip3.
Now I removed python 3.9 with brew and deleted with rm all those files.
